### ISSUE TITLE
klarna payment no email or empty emails received

### ISSUE THREAD

**Issue by Spiros333** [mollie/PrestaShop#358]
Expected behavior

klarna payment authorized should be klarna payment authorized status

Actual behavior

Order page shows klarna payment authorized but if you look in the order it says Awaiting Mollie payment (see screenshots)

We receive no confirmation email of the purchase.
The customer doesn't receive confirmation email of his purchase.
The customer doesn't receive email att with his invoice.

Steps to reproduce the behavior
pay later order received from customer

Environment

    PrestaShop version: 1.7.6.2
    Mollie module version: 4.3.0
    PHP version: 7.2

![afbeelding](https://user-images.githubusercontent.com/85503636/121754332-a13f5980-cb14-11eb-839d-7dc7846e7d60.png)

![afbeelding](https://user-images.githubusercontent.com/85503636/121754313-95ec2e00-cb14-11eb-86b1-a6cd69b2de69.png)

**Comment by Spiros333** [comment]
PLEASE DON"T MAKE WITH EVERY MOLLIE RELEASE THAT THE KLARNA STATUSSES ARE DELETED AND RECREATED WITH A NEW STATUS ID!!!! THIS BRINGS EVERYTIME PROBLEMS
COULD YOU PLEASE ARRANGE THIS?

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

Thank you for your message.

I understand your frustration regarding this case, but I've tested the case several times, and unfortunately I could not reproduce the same issue, seems that the email arrives and it looks as it should (attaching screenshots):

![image](https://user-images.githubusercontent.com/75423142/121863709-ce525e80-cd04-11eb-9d7b-8ac17a40af51.png)
![image](https://user-images.githubusercontent.com/75423142/121863733-d3afa900-cd04-11eb-9274-479a7a28ba07.png)
![image](https://user-images.githubusercontent.com/75423142/121863753-d7dbc680-cd04-11eb-8b19-2dc9f0a90f48.png)


Maybe you could provide us with your BO+FTP credentials to support@invertus.eu and we could check if everything is okay with your shop?

Looking forward to your reply!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
I know things are working fine but why when updating mollie everytime klarna payment authorized status deleted & create new one status with new id codes?

My klarna payment authorized status was before updating mollie 117 ID now it is 123ID can you please stop this?
Don't let it create new id's please
![afbeelding](https://user-images.githubusercontent.com/85503636/121886136-58a2be80-cd15-11eb-8b32-7657bbc86e56.png)

**Comment by Spiros333** [comment]
Can you please hardcode that when the status Klarna Payment Authorized and Klarna payment shipped exists on prestashop as satus to not recreate them with a new id code?

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

Resetting the module creates the new statuses, if you do not reset the module, after update the statuses should not change, as I've just tested the same scenario from 4.2.4 to 4.3.0. Hardcoding in this case is not an option, because other customers could be frustrated as the statuses might not be removed when the module is deleted, or they might start duplicating.

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
The module was not resetted 

When updating mollie you recomment to uninstall the old version and install the new one.

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

And could you clarify if you've tried just to install the new version on top of the old one, and if everything was okay?

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
new version on top of the old one, 
>>No received an error so deleted the old one and installed the new one.

**Comment by Saimis777** [comment]
Thanks @Spiros333 ,

And also, what is the error that you've received when trying to install the new version could you clarify please?

Thank you!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
![afbeelding](https://user-images.githubusercontent.com/85503636/122552161-3e7b2000-d036-11eb-88d2-bdeb8e2b7bf3.png)

**Comment by Spiros333** [comment]
Oops whoops failed again try a professional payment provider.
Sorry but this is really a joke.
Why are your plugins so bad?

**Comment by SimonasB88** [comment]
Hello dear @Spiros333 
And what is the message when "What happened?" is clicked?

Best Regards
Invertus Support Team

**Comment by Spiros333** [comment]
Error

**Comment by SimonasB88** [comment]
@Spiros333 there could be hundreds of errors referring to the all Prestashop modules.
Could you show the error text?

Best Regards
Invertus Support Team

**Comment by Saimis777** [comment]
Hello @Spiros333 ,

Would it be possible to receive more details?

Looking forward to your reply!

--
Best regards,
Invertus Support team

**Comment by pic16f876** [comment]
Hello,

I also had issues with Mollie 4.2.3 and Klarna, but realised this very late because Klarna is not used often on our site. I updated today to 4.3.0 and will keep you updated if the error reoccurs. Once the customer did not receive the email and on another occasion, the customer was told the adress was wrong. Both cases valid adresses and emailadresses. 

Presta 1.7.6.5
PHP 7.3
Mollie 4.2.3

**Comment by Spiros333** [comment]
Ive it also with 4.3.0

**Comment by Saimis777** [comment]
Hi @pic16f876 ,

Thank you for your feedback, let us know if 4.3.0 will resolve the issue for you!

@Spiros333 Maybe you could press the "what happened?" button and screenshot the further error messages that it showed for you please?

Looking forward to your reply!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
@pic16f876 Do you think I stuppid to try this ones again.
It took me hours to get mollie working, test it on your own.
Whats wrong with this product nothing is customer friendly and ones again to put all the settings in by hand.

**Comment by pic16f876** [comment]
@Saimis777 Could take a while, as allmost everyone uses ideal, bancontact or credit card with us. But will definately keep you posted. If you really want to know you can order a nice perfume with Klarna on our website ;)

@Spiros333 I am not sure what you try to tell me. I just wanted to add my experience and I am not telling you do do anything. I allways do a full reinstall with every update and reconfigure by hand, which is quite cumbersome indeed. But better to reconfigure by hand (takes me 10 minutes now) then destroying the BO (what happens when you update the module allmost every time)  and recovering from a backup. But I do agree with you that a reinstall should not be necessary everytime.

**Comment by Spiros333** [comment]
@pic16f876  excuse me my commend was ment as reaction for @Saimis777

**Comment by pic16f876** [comment]
@Saimis777 Just received a klarna payment in Live. All went well. As mentioned above I did a clean install of 4.3.0

**Comment by Saimis777** [comment]
@pic16f876 Thanks for the clarification! Glad that everything works!

@Spiros333 I am really sorry for the inconveniences I understand it's frustrating for you, regarding the update issue, the development team is working on it, and I will clarify the further causes.

Thank you for your patience!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Dear Invertus Support team,

Question 

1. Can I make klarna apear as option only when the customer purchase above the 100 euros?
2. Can I make paypal disapear when the purchase is above the 400?

Please let me know.
Thanks

**Comment by petrovv77** [comment]
@Spiros333 Not with this module. You should play with activating / deactivating carriers at a certain amount and possibly run paypal and Klarna as a separate module on its own and have them activated through the carrier at a certain amount.

But they may want to build it in in the future. It is not a basic function, but it is certainly useful. Good question!

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

Basically, it's the same functionality. At the moment we do not provide this functionality to disable/enable payments from the certain price, but the job for this functionality is estimated, and is now under Mollie review, so I will definitely update you once we will know if this functionality will be approved or not!

Thank you for your patience!

--
Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hello @Spiros333 ,

Thank you for your patience!

I've got confirmation from Mollie that we will do an update to save the statuses, and not remove them. We are aiming to release this change in the following release after 4.3.1!

Please let me know if you have any further questions in the meantime!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Invertus how can this be...?

an pay later order still open of this morning 2021-07-07 08:57:10 Status awaiting mollie payment.
This should get cancelled after some time?
This customer now thinks his order was placed and completed because they didn't received an cancellation email


![afbeelding](https://user-images.githubusercontent.com/85503636/124837633-7ca89700-df85-11eb-9332-b8688715d9f9.png)

![afbeelding](https://user-images.githubusercontent.com/85503636/124837747-c1ccc900-df85-11eb-9269-5287b5898b61.png)

**Comment by Spiros333** [comment]
I see it already that its mentioned in the mollie dashboard, it will be cancelled after one day

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

If you use payment API then it will expire after the time it shows here: https://docs.mollie.com/payments/status-changes 
So Klarna 48h.

If you are using order API it will expire after 28days. This is done by Mollie API not by module.

Let me know if you have any more questions!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Ive updated to the newest version 4.3.2.

Your plugin again deleted my old status IDs and created new ones.

statusses 4.3.0
![afbeelding](https://user-images.githubusercontent.com/85503636/125201830-17002780-e271-11eb-98f4-83b533bee308.png)

Statusses 4.3.2
![afbeelding](https://user-images.githubusercontent.com/85503636/125201855-3a2ad700-e271-11eb-82e3-2837467f5dab.png)

**Comment by Spiros333** [comment]
it changed also my carrier id DHL, the ID number was 176 and now its 201.

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

The fix has not been implemented yet, 4.3.2 was released due to a small fix that our team has come across with to be changed, and replaced the 4.3.1 version,. Once the fix will be implemented, and released, I will inform you personally. We have added the fix to the next release.

Thank you for your patience, and collaboration!

Let me know if you have any more questions in the meantime!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Still problem with klarna orders.
Me and my customers do not receive confirmation emails when an purchase is done.
How long will this take as a problem?
Could you please login and fix this problem?

**Comment by Saimis777** [comment]
Hey @Spiros333 ,

I'm really sorry, but the issue could not be reproduced on a default installation, maybe it would be possible to receive your credentials, and we would gladly check the issue for you! You would need to send them to - support@invertus.eu

Let me know if that would be possible!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
just sent it out to support@invertus.eu

**Comment by Spiros333** [comment]
Simonas from Invertus Support team found an fix to my problem.

We fixed the issue for you, there was an issue with validators where klarna authorized status was not checked as paid. 

I will let you know if this fix made some diffrence

**Comment by Saimis777** [comment]
Hello,

We fixed an issue when the status of a klarna payment changed to klarna authorized, and the email sending validation did not read it as "paid", and therefore it did not send the email confirmation to the merchant, now it is fixed, and the email is sent correctly.

Let us know if there are any further questions.

--
Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hey @Spiros333 ,

The new 4.3.3 release is now out! You can download it here - https://github.com/mollie/PrestaShop/releases/tag/4.3.3

It contains the new feature I've mentioned before, that it will no longer reset the ID's, as well as the fix for email confirmations.

I will mark the issue for now as resolved.

If you will need further help, let me know, and I will re-open it!

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Hi thanks,

Do I need to delete the previous module before installing this one?
Please let me know

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

No need to delete, you can update the current version, we've tested many times upgrade from 4.3.2 to 4.3.3. everything seemed to work as it should, without any errors.

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
It works fine

1. I really want to be able to enable and disable the payment methods per catagory.
SO that for example paypal is not available for my catagory "wholesales"

2. I want to be able to put per payment methode an Min and an Max totaal price.
SO for example klarna is only evalable above the 100euro.
And Paypal is not available above the 900euro as example.

Could you give us in the plugin multiple option possibilitys were we could deccide the control of the payment methods?

Please let me know

**Comment by Spiros333** [comment]
Just received an klarna payment order YPBWGMXIW #7765
We and the customer didn't receive an confirmation email AGAIN
Can you again look into this please?

I want a deffently solution for this!!!

**Comment by Spiros333** [comment]
The problem is back again after installing 4.3.3. 

SO I had updated the module to 4.3.3.  and after 40min later we received an klarna payment but no email

**Comment by Saimis777** [comment]
@Spiros333 ,

For any feature requests/more plugin possibilities, please raise a new issue in GitHub.

For the klarna payment no email part, I've forwarded the information to the developers, waiting on their response.

Thank you for your patience.

--
Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
@Spiros Panagiotopoulos ***@***.***> ,

Can we create live test orders to check if everything is okay, and emails
are being sent?

Looking forward to your reply!

--
Best regards,
Invertus Support team

**Comment by spiros** [comment]
I think you have the wrong @spiros here - CC @Spiros333

**Comment by Spiros333** [comment]
Okay make it clear in the order that its a test

Verstuurd vanaf mijn iPhone

**Comment by Saimis777** [comment]
Sorry about that, yes, the message was meant for @Spiros333

**Comment by Saimis777** [comment]
@Spiros333 ,

I've sent you an email regarding the case.

--
Best regards,
Invertus Support team

**Comment by Spiros333** [comment]
Goodafternoon,

It happend again just moments ago with a order of a customer again with paylater klarna, its order ZXXCCBZIF #7790.

We and the customer didn't receive an email as confirmation about the order.

Keep in mind, the FTP password is changed see email.

Please let me know what is happening and when this finally will be fixed!!!

 13-08-2021 13:18 Simonas Burneikis <simonas.burneikis@invertus.eu>:


Hello Spiros,

So developer has fixed Sentry implementation in module. He suggests you to reproduce the issue again - step-by-step, then we will check what Sentry logs are registering from module side.

Could you reproduce it once again?

Best Regards.

**Comment by Spiros333** [comment]
can you reopen this, still not solved.

**Comment by Saimis777** [comment]
Hi @Spiros333 ,

Let us know if you need further help regarding this case.

Closing it for now.

--
Best regards,
Invertus Support team
